src/test/groovy/graphql/execution/directives/QueryDirectivesImplTest.groovy

```diff
@@ -87,6 +86,28 @@ class QueryDirectivesImplTest extends Specification {
 
         then:
         appliedDirectivesByName.keySet().sort() == ["cached", "timeout", "upper"]
+    }
+
+    def "gets repeated definitions"() {
+        def f1 = TestUtil.parseField("f1 @rep(place: \$var) @rep(place: \"HELLO\")")
 
+        def mergedField = MergedField.newMergedField([f1]).build()
+
+        def queryDirectives = QueryDirectives.newQueryDirectives()
+                .mergedField(mergedField)
+                .schema(schema)
+                .coercedVariables(CoercedVariables.of([var: "ABC"]))
+                .graphQLContext(GraphQLContext.getDefault())
+                .locale(Locale.getDefault())
+                .build()
+
+        when:
+        def appliedDirectivesByName = queryDirectives.getImmediateAppliedDirectivesByName()
+
+        then:
+        appliedDirectivesByName.keySet() == ["rep"] as Set
+        appliedDirectivesByName["rep"].size() == 2
+        // Groovy is a pathway to many abilities some consider to be unnatural
```

Comment:
> Groovy is a pathway to many abilities some consider to be unnatural
Lol

Reply:
My favourite is groovy will ignore all 'private' modifier. I am not selfish. I can even share other's private with you.